.github/workflows/lint-codeql.yaml

```diff
@@ -23,10 +23,19 @@ jobs:
         uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579
         with:
           persist-credentials: false
+      - name: Retrieve pull request's base and head
+        if: ${{ github.event.issue.pull_request }}
+        id: pr
+        run: |
+          curl ${{ github.event.issue.pull_request.url }} > pr.json
+          echo "::set-output name=base::$(jq -r '.base.sha' pr.json)"
+          echo "::set-output name=head::$(jq -r '.head.sha' pr.json)"
       - name: Check code changes
         uses: dorny/paths-filter@b2feaf19c27470162a626bd6fa8438ae5b263721
         id: go-changes
         with:
+          base: ${{ steps.pr.outputs.base }}
+          ref: ${{ steps.pr.outputs.head }}
```

Comment:
For a workflow run triggered by a pull request, I think `github.event.pull_request.{head,base}.sha` are enough here (`github.event.issue` is probably not available on that trigger anyway). But this workflow can also be triggered on schedule on by push to the master branch; shouldn't we account for that as well in these values?

Reply:
Good point. I guess we only want to compare against the base for `github.event_name == pull_request` (i.e. to only check code changes), but run against the full code base on scheduled runs. The dorny/paths-filter action seems to require `base` or `repository.default_branch` being set for scheduled runs as well, thus the error. So I guess we just make that action conditional on `github.event_name == pull_request`.

Reply:
> So I guess we just make that action conditional on `github.event_name == pull_request`.

Yep, I think that would be fine.